tsnet/tsnet.go

```diff
@@ -238,15 +240,15 @@ func (s *Server) start() error {
 	}
 	logid := lpc.PublicID.String()
 
-	f, err := filch.New(filepath.Join(s.rootPath, "tailscaled"), filch.Options{ReplaceStderr: false})
+	s.logbuffer, err = filch.New(filepath.Join(s.rootPath, "tailscaled"), filch.Options{ReplaceStderr: false})
```

Comment:
missing cleanup in error paths below, like monitor.New failing

Reply:
Maybe time to reuse:

```go
	var closePool closeOnErrorPool
	defer closePool.closeAllIfError(&reterr)
```

From wgengine?

Reply:
There are a number of resource leaks on the error path here. I'll fix that in a subsequent PR since this PR is clearly an improvement over the status quo.